### PR TITLE
Navigation manager enhancement

### PR DESCRIPTION
**Pull request by TheImpressiveDonut**

### CODE REVIEW

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java [hunk 1]
```diff
@@ -42,44 +61,59 @@ public NavigationManager(Context context, DeviceLocationProvider deviceLocationP
         Objects.requireNonNull(context);
         Objects.requireNonNull(deviceLocationProvider);
         this.mapboxMap = mapboxMap;
-        deviceLocationProvider.addObserver((subject, value) -> {
-
-        });
+        this.context = context;
+        this.routeDisplayManager = routeDisplayManager;
```
Comment (Sosol02) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java:65]: Maybe check that Objects.requireNonNull(routeDisplayManager)?

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java [hunk 2]
```diff
@@ -96,31 +130,73 @@ public void onLocationObservationReceived(@NonNull LocationObservation locationO
...
+
+        }
+    }
+
+    private class UpdatingSpeedLimitSpanListener implements SpeedLimitSpanListener {
+
+        @Override
+        public void onSpeedLimitBoundariesCrossed(@NonNull Set<SpeedLimitSpan> exitedSpeedLimits, @NonNull Set<SpeedLimitSpan> enteredSpeedLimits) {
+            for (SpeedLimitSpan speedLimitSpan : enteredSpeedLimits) {
+                float speed = speedLimitSpan.getSpeedLimit().getSpeed();
```
Comment (Sosol02) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java:179]: Is this unused?

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java [hunk 3]
```diff
@@ -96,31 +130,73 @@ public void onLocationObservationReceived(@NonNull LocationObservation locationO
...
+                SpeedLimit.Type type = speedLimitSpan.getSpeedLimit().getType();
```
Comment (Sosol02) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java:180]: Same here, looks useless.

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java [hunk 4]
```diff
@@ -96,31 +130,73 @@ public void onLocationObservationReceived(@NonNull LocationObservation locationO
...
+            }
+        }
+    }
+
+    private class EtaUpdateResponseListener implements EtaResponseListener {
 
+        @Override
+        public void onEtaUpdate(@NonNull EstimatedTimeOfArrival estimatedTimeOfArrival) {
+            long timeLeft = estimatedTimeOfArrival.getTime();
```
Comment (Sosol02) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java:189]: That does nothing as well.

**Review by Sosol02 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java (hunk 1 above)
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java:65]: True

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java (hunk 4 above)
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java:189]: Yeah, I use it in the next pr for navigation UI

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java (hunk 2 above)
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/navigation/fragment/map/NavigationManager.java:179]: Yeah, I use it in the next pr for navigation UI

**Review by Sosol02 — APPROVED** [review]

File: app/src/androidTest/java/com/github/onedirection/map/MapFragmentTest.java [hunk 5]
```diff
@@ -355,7 +356,7 @@ public void testRoutesManagerFindMethod() throws InterruptedException {
         setAttributeField("routeService", routesManager, routeService);
         Semaphore semaphore = new Semaphore(0);
```
Comment (justagoodguy-omega) [inline app/src/androidTest/java/com/github/onedirection/map/MapFragmentTest.java:357]: Why do you use semaphores in the code? :)

**Review by justagoodguy-omega — APPROVED** [review]

File: app/src/androidTest/java/com/github/onedirection/map/MapFragmentTest.java (hunk 5 above)
Comment (TheImpressiveDonut) [reply app/src/androidTest/java/com/github/onedirection/map/MapFragmentTest.java:357]: I tried using idling resource but i doesn't seem to work well in those cases. So I decided to use semaphore to wait for the function to finish, it works pretty well too, it's just a bit uglier than idling resource.

Comment (TheImpressiveDonut) [reply app/src/androidTest/java/com/github/onedirection/map/MapFragmentTest.java:357]: I tried using idling resource but i doesn't seem to work well in those cases. So I decided to use semaphore to wait for the function to finish, it works pretty well too, it's just a bit uglier than idling resource.

**Pull request merged**